monai/apps/pathology/transforms/spatial/array.py

```diff
@@ -73,3 +75,133 @@ def get_params(self, image_size):
         )
 
         return patch_size, steps
+
+
+class TileOnGrid(Transform):
+    """
+    Tile the image into patches on a grid and maintain a subset of it.
+    This transform works only with np.ndarray inputs.
+
+    Args:
+        tile_count: number of tiles to extract
```

Comment:
As this transform name is `TileOnGrid`, I think we don't need the prefix "tile_" in every arg name?

Thanks.

Reply:
I like it more this way.  "tile_size" is more explicit than var "size", and "tile_count" is more descriptive than just "count".   Is there a reason we can't keep it this way?

Reply:
Hi @ericspod @wyli ,

What do you guys think about this naming here?

Thanks in advance.

Reply:
I think it's better to have more information in the argument names but if they all have the same prefix it doesn't add any new information about what they are or do. If the prefix was used to differentiate arguments (eg. different size parameters) it would make sense to have a prefix for each, but here if you take away `tile_` nothing is lost because the context of the arguments being related to tiles is still present. I would change `tile_all` to `extract_all` for example, that states the verb and the noun related to what's going on.

Reply:
Thanks for participating in our silly discussion. but I still prefer my names.  Tile is also a verb (to tile or to extract).   For this particular argument I don't mind changing it, but I also don't see a reason for it. For other arguments  "tile_count" is much more descriptive than "count", or "tile_size" is much more informative than "size". It's okay to have redundancy in names, redundancy help to avoid potential bugs.  If I have to search (in the whole Monai project for tile_count  it will immediately find this function,  if I have to search for keyword "count" or "extract_count" to see where else it's used, it may return me a long list of results, that's hard to analyze.

Reply:
Since we are on the topic of multiple instance learning, why don't we use the widely used terms such as bags and instances? Why do we introduce patch/grid/tile? To me these are unnecessarily confusing

Reply:
this transform is generic, and can be used outside of MIL.  bags and instances will be very confusing terms here.  this transform simply splits the image into square patches.  to split or to tile both sounds like good names, but "split" is already taken

Reply:
Picking the top n instance out of a bag is already implying the mil assumption. 

Also, tile means a completely different thing in numpy, this transform happens to handle only numpy, it is confusing.

Reply:
so do you want to rename the transform? or rename the arguments? 
please suggest the names

Reply:
Perhaps "ImageToBag"? Converting an image to a set of patch instances based on a spatial grid sampling strategy. Then we can have some parameters to define grid properties and the size of a bag.

Reply:
alright, I think we found a comporimise